R/carer_payment.R

```diff
@@ -0,0 +1,116 @@
+                          children_annual_income = 0,
+                          children_asset_value = 0
+                          ) {
+  
+  if (is.null(Date)) {
+    if (is.null(fy.year)) {
+      Date <- .age_pension_today2qtr(Sys.Date())
+      fy.year <- date2fy(Date)
+      message('`Date` and `fy.year` not set, so using `Date = "', Date, '".')
+    } else {
+      Date <- fy2date(fy.year)
+    }
+  } else {
+    if (is.null(fy.year)) {
+      fy.year <- date2fy(Date)
+    } else {
+      warning("`fy.year` and `Date` both used. Ignoring `fy.year`.")
+    }
+  }
+  
+  #error for no dclad or adat
+  
+  input <- data.table(do.call(cbind.data.frame, mget(ls())))
+
+  #Rates, income test, and asset test same as age pension
+
+  rate <- age_pension(ordinary_income = carer_fortnightly_income,
+                      annual_income = carer_annual_income,
+                      has_partner = carer_has_partner,
+                      n_dependants = carer_n_dependants,
+                      partner_fortnightly_income = carer_partner_fortnightly_income,
+                      partner_annual_income = carer_partner_annual_income,
+                      Date = Date,
+                      assets_value = carer_assets_value,
+                      is_home_owner = carer_is_home_owner,
+                      illness_separated_couple = carer_illness_separated_couple)
+
+  
+
+  #http://guides.dss.gov.au/guide-social-security-law/4/2/5
+  #https://www.humanservices.gov.au/sites/default/files/co029-1603en.pdf
+  
+  #detailed eligibility: http://guides.dss.gov.au/guide-social-security-law/3/6/4
+  care_receiver_eligible <- input[ ,(dclad_eligible | high_adat | low_adat)]
+
+  care_receiver_income_test <- input[ ,if_else(high_adat,
+                                               care_receiver_annual_income + partner_annual_income, #high adat
+                                               if_else(dclad_eligible & living_at_home,
+                                                       care_receiver_annual_income, #child living away from home
+                                                       care_receiver_annual_income + partner_annual_income + children_annual_income))] #all other cases
+  
+  care_receiver_asset_test <- input[ ,if_else(dclad_eligible & !living_at_home,
+                            care_receiver_asset_value, #child living away from home
+                            care_receiver_asset_value + partner_asset_value + children_asset_value)] #all other cases
+
+  income_eligible <- input[ ,if_else(high_adat & receiving_other_payment, #exemption: http://guides.dss.gov.au/guide-social-security-law/3/6/4/10
+                                      TRUE,
+                                      (care_receiver_income_test < 108828))]
+  assets_eligible <- input[ ,if_else(high_adat & receiving_other_payment,
+                                      TRUE,
+                                      (care_receiver_asset_test < 671250))]
```

Comment:
These are both "or" statements.

Reply:
You sure? http://guides.dss.gov.au/guide-social-security-law/3/6/4/10 under Income & Assets Testing says that those with high ADAT and receiving payment are exempt from asset and income testing.

Reply:
I meant the `if_else`: 

```r
assets_eligible <- input[, or(high_adat & receiving_other_payment, care_receiver_asset_test < 671250)]
```